src/hooks/wallets/useInitWeb3.ts

```diff
@@ -26,30 +27,33 @@ export const useInitWeb3 = () => {
   const wallet = useWallet()
   const customRpc = useAppSelector(selectRpc)
   const customRpcUrl = chain ? customRpc?.[chain.chainId] : undefined
-
-  useEffect(() => {
-    if (wallet && wallet.chainId === chainId) {
-      const web3 = createWeb3(wallet.provider)
-      setWeb3(web3)
-    } else {
-      setWeb3(undefined)
-    }
-  }, [wallet, chainId])
+  const web3 = useWeb3()
 
   useEffect(() => {
     if (!customRpcUrl || !chainId) {
-      setWeb3ReadOnly(undefined)
-      setMultiWeb3ReadOnly(undefined)
-
-      if (chain && router.pathname !== AppRoutes.welcome.index && router.pathname !== AppRoutes.index) {
+      if (!wallet && !web3) {
+        setWeb3(undefined)
+        return
+      }
+      if (wallet && wallet.chainId !== chainId) {
         dispatch(
           showNotification({
-            message: `No RPC URL saved for ${chain.chainName ?? 'this'} network. You must provide one to continue.`,
+            message: `Your wallet seems to be connected to the wrong network. You must change your wallet network to ${
+              chain?.chainName ?? 'the same network'
+            }.`,
             groupKey: RPC_URL_ERROR_KEY,
             variant: 'error',
           }),
         )
-        router.push({ pathname: AppRoutes.welcome.index, query: { chain: chain.shortName } })
+        return
+      }
+      if (wallet && wallet.chainId === chainId) {
+        const internalWeb3 = createWeb3(wallet.provider)
+        setWeb3(MulticallWrapper.wrap(internalWeb3, 50))
+        if (internalWeb3) {
```

Comment:
Can you add `dispatch(closeByGroupKey({ groupKey: RPC_URL_ERROR_KEY }))` above this line please so that the "Your wallet seems to be connected to the wrong network" notification is automatically dismissed after the user does set their wallet to the right network? Thanks!